CPU_consumption/Consumption/CoresConsumption.cpp

```diff
@@ -0,0 +1,61 @@
+//
+// Created by Nikolay Fedotenko on 03.03.2021.
+//
+
+#include "CoresConsumption.h"
+
+/*! \file
+ *  This source defines CoresConsumption class
+ */
+
+CoresConsumption::CoresConsumption(int         n,
+                                   const Rect& currentArea)
+{
+    mach_msg_type_number_t count;
+    processor_info_array_t coreInfo;
+    host_processor_info(mach_host_self(),
+                        PROCESSOR_CPU_LOAD_INFO,
+                        &_coreCount,
+                        &coreInfo,
+                        &count
+    );
+
+    _area = currentArea;
+    _number = n;
+    _cores = (core_info_t*)malloc(sizeof(core_info_t) * _coreCount);
+
+    initializeShaderData();
+}
+
+CoresConsumption::~CoresConsumption()
+{
+    free(_cores);
+}
+
+double CoresConsumption::getCurrentValue()
+{
+    mach_msg_type_number_t count;
+    processor_cpu_load_info_t coreInfo;
+    natural_t user, system, idle, nice;
+    double used, total;
+
+    host_processor_info(mach_host_self(),
+                        PROCESSOR_CPU_LOAD_INFO,
+                        &_coreCount,
+                        (processor_info_array_t*)&coreInfo,
+                        &count
+    );
+    user = coreInfo[_number].cpu_ticks[CPU_STATE_USER] - _cores[_number].cpu_ticks[CPU_STATE_USER];
+    system = coreInfo[_number].cpu_ticks[CPU_STATE_SYSTEM] - _cores[_number].cpu_ticks[CPU_STATE_SYSTEM];
+    idle = coreInfo[_number].cpu_ticks[CPU_STATE_IDLE] - _cores[_number].cpu_ticks[CPU_STATE_IDLE];
+    nice = coreInfo[_number].cpu_ticks[CPU_STATE_NICE] - _cores[_number].cpu_ticks[CPU_STATE_NICE];
```

Comment:
Geez
Maybe extract to separate method?
There's A LOT of similar code in here...